### ISSUE TITLE
Restart of daily production

### ISSUE THREAD

**Issue by patman15** [SonnenladenGmbH/APsystems-EZ1-API-HomeAssistant#34]
Hi!

I'm having troubles with my statistics because the daily production counter restarts when the sun goes away and comes back. Mostly happens in the evening but also on very cloudy days. This causes further calculations that depend on that sensor to jump around. I would suggest to fix this by adding up the values of the inverter and reset only once at midnight. What do you think?
As previously, I can also suggest a fix (PR) if you want me to.

See the issue here:
![daily_production](https://github.com/SonnenladenGmbH/APsystems-EZ1-API-HomeAssistant/assets/14628713/2cb66cc1-0912-4619-9d18-3b9055f71ebf)

**Comment by mawoka-myblock** [comment]
Thanks for the offer, but I want this repo to get merged into HomeAssistant Core, so please don't open the PR here, but the PR is stuck at the moment, so I don't have a clue when this will get merged. You can check the PR here: https://github.com/home-assistant/core/pull/114531

**Comment by patman15** [comment]
Ok understood, I'll do some local fixes and hope for the best for your PR :crossed_fingers:. Ping me if you want me to provide the fixes.

**Comment by mawoka-myblock** [comment]
Once it's merged, you can submit the fixes directly to HA. But I'm not sure the PR will ever get merged.

**Comment by patman15** [comment]
> But I'm not sure the PR will ever get merged.

I had a look at the code, are you sure that all prerequisites like the coding style are fulfilled? I had a quick look and my linters (ruff, pylint) show a number of issues.

**Comment by mawoka-myblock** [comment]
Will check that. Could be that the merges have messed sth up. It's been some weeks since I've checked everything.

**Comment by mawoka-myblock** [comment]
I've now fixed a new linter warning. The rules has been enabled in the last few weeks. Everything passes now.
